tensorflow/java/src/test/java/org/tensorflow/TensorTest.java

```diff
@@ -401,71 +479,74 @@ public void failCreateOnMismatchedDimensions() {
   }
 
   @Test
-  public void failCopyToOnIncompatibleDestination() {
+  public void whenCopyToWithDimensionMismatchThenThrowExpectedException() {
```

Comment:
Many of these names seem more verbose than they need to be. Is there some well known convention around naming all tests as "whenXThenY" that you're following? I'd be tempted to stick with more succinct names.